infra/templates/az-isolated-service-single-region/tests/integration/integration_test.go

```diff
@@ -62,6 +66,42 @@ func httpGetRespondsWith200(goTest *testing.T, output infratests.TerraformOutput
 	}
 }
 
+// Verifies that the Key Vault instance deployed is properly isolated within the VNET
+func verifyVnetIntegrationForKeyVault(goTest *testing.T, output infratests.TerraformOutput) {
+	vaultResourceGroup := output["app_dev_resource_group"].(string)
+	vaultName := output["keyvault_name"].(string)
+	keyVaultACLs := azure.GetKeyVaultNetworkAcls(goTest, adminSubscription, vaultResourceGroup, vaultName)
+	subnetIDs := azure.GetSubnetsInVnet(goTest, adminSubscription, aseResourceGroup, os.Getenv("TF_VAR_ase_vnet_name"))
+
+	// No azure services should have bypass rules that allow them to circumvent the VNET isolation
+	if keyVaultACLs.Bypass != keyvault.None {
+		goTest.Fatal(fmt.Errorf("Expected bypass option of %s but got %s", keyvault.None, keyVaultACLs.Bypass))
+	}
+
+	// The default action should be to deny all traffic
+	if keyVaultACLs.DefaultAction != keyvault.Deny {
```

Comment:
you may want to consider using `assert.Equal` here from `github.com/stretchr/testify/assert`.